libraries/rush-lib/src/cli/RushXCommandLine.ts

```diff
@@ -45,129 +47,139 @@ interface IRushXCommandLineArguments {
 }
 
 export class RushXCommandLine {
-  public static launchRushX(launcherVersion: string, isManaged: boolean): void {
-    RushXCommandLine._launchRushXInternal(launcherVersion, { isManaged });
+  public static launchRushX(launcherVersion: string, options: ILaunchRushXInternalOptions): void {
+    try {
+      const args: IRushXCommandLineArguments = this._getCommandLineArguments();
+      const rushConfiguration: RushConfiguration | undefined = RushConfiguration.tryLoadFromDefaultLocation({
+        showVerbose: false
+      });
+      const eventHooksManager: EventHooksManager | undefined = rushConfiguration
+        ? new EventHooksManager(rushConfiguration)
+        : undefined;
+
+      const ignoreHooks = process.env.INSIDERUSH === '1';
+      eventHooksManager?.handle(Event.preRushx, false /* isDebug */, ignoreHooks);
+      RushXCommandLine._launchRushXInternal(launcherVersion, options, rushConfiguration, args);
+      eventHooksManager?.handle(Event.postRushx, false /* isDebug */, ignoreHooks);
```

Comment:
Event hooks are called here because the multiple exit points in _launchRushXInternal requires code duplication to handle each return case.   Looking around the code, I couldn't see any compelling reasons for external vs internal, so I moved setup/teardown work into this function.  This ultimately makes the internal version easier to test.